### ISSUE TITLE
Adjusted pictures in repeated block cause document to become unreadable

### ISSUE THREAD

**Issue by Tristanhx** [Skrol29/opentbs#26]
Hello!
We have a problem with documents becoming unreadable when pictures in a repeating block (sub-block) configuration are adjusted.

![image](https://user-images.githubusercontent.com/27008795/169979842-b5952660-4534-4a83-b683-360160a1d2f7.png)
I have a template like the above image where this block needs to be repeated an amount of times. Inside is a sub-block that contains a picture. This placeholder picture is replaced by a picture from an array like so: 
```
[
    "findings" => [
        [
            "proofsegments" => [
                [
                    "image" => <pathtoimage>,
                ],
                [
                    "image" => <pathtoanotherimage>,
                ],
            ]
        ],
        [
            "proofsegments" => [
                [
                    "image" => <pathtoimage>,
                ],
                [
                    "image" => <pathtoanotherimage>,
                ],
            ]
        ],
    ],
];
```

This works fine as long as the image is not adjusted. To prevent the image potentially being wider than the page we use adjust=samewidth to make the image taller as necessary. This produces an unreadable document. The reason for this is twofold.

1. The new height that is supposed to replace the old height in the xml replaces another portion of the xml
`<a:ext cx="5788152" cy="3090672"/></a:xfrm><a:prstGeom prst="rect"><a:avLst/></a:prstGeom>6259281pPr></pic:pic>`

It should be that the value 6259281 replaces the value of cy attribute in the <a:ext/> tag.

2. The NewRid is also misplaced, breaking the neighboring xml tag.
`<a:ext uri="{28A0092B-C50C-407E-A947-70E740481C1Copentbs114:useLocalDpi xmlns:a14="http://schemas.microsoft.com/office/drawing/2010/main" val="0"/>`

It seems the misplaced alterations always are placed too far to the right. Adjusting pictures works fine when not in a repeating block.
In the function TbsPicAdjust the new height is replacing the old one in $Txt via `$Txt = substr_replace($Txt, $new, $beg, $len);` It appears that $beg now has a value that is too large causing the replacement to occur too far to the right. In my specific case subtracting 66 from $b placed the new height at the correct spot, but in another test with a different template document without sub-blocks the height was misplaced by a different amount. I suspect a similar thing is happening in TbsPicExternalPath or TbsPicAdd.

Thank you for your time!

**Comment by Tristanhx** [comment]
I have found a work around, which is to inject TBS syntax instead of a picture. This way the repeated blocks are already executed before merging any pictures. I injected `[onshow.picturename;ope=changepic;tagpos=inside;adjust=samewidth]` into the description of the images. "picturename" is a randomly generated name that corresponds to a key/value pair in $VarRef so I can add the pictures during Show. The injected syntax will be partly encoded. `[onshow` becomes `&#91;onshow`. I had to replace those occurrences with the non-encoded version by altering the Show function and altering $Source: 
```
public function Show($Render=false) {
	//have to do this to fix injected tbs syntax
	$this->Source = str_replace("&#91;onshow", "[onshow", $this->Source);
	$Ok = true;
	[...]
	return $Ok;
}
```

This is an ugly hack and it would be better if adjusting pictures that are in repeating blocks could be done during the merging of that block.

**Comment by drentjess** [comment]
I am having the exact same issue with a block section where I want to do a changepic and add the adjust parameter. Without the adjust parameter it is working fine, but then as expected my images are not adjusted nicely. I want to have the images adjusted within the block section. I would prefer a different fix then the one mentioned here, but then again I appreciate the efforts!

**Comment by GarciaSiego** [comment]
I figured out what the offset issue is for this, its the description tag being emptied. In TbsPicGetDim_OpenXML_dml it gets the offset, but its with the description still filled with for example:
`[opportunities.testje_field_sugar_image;ope=changepic;tagpos=inside;adjust]`
And in TbsPicAdjust we can see the $Txt description now being empty.

So if we now adjust $beg length to subtract the total length of the old value of description, it works. 

Made snippet as example:
`$text ='</w:t></w:r></w:p><w:tbl><w:tblPr><w:tblW w:w="9004" w:type="dxa"/><w:tblInd w:w="108" w:type="dxa"/><w:tblBorders><w:top w:val="single" w:sz="8" w:space="0" w:color="FFFFFF"/><w:left w:val="single" w:sz="8" w:space="0" w:color="FFFFFF"/><w:bottom w:val="single" w:sz="8" w:space="0" w:color="FFFFFF"/><w:right w:val="single" w:sz="8" w:space="0" w:color="FFFFFF"/><w:insideH w:val="single" w:sz="8" w:space="0" w:color="FFFFFF"/><w:insideV w:val="single" w:sz="8" w:space="0" w:color="FFFFFF"/></w:tblBorders><w:shd w:val="clear" w:color="auto" w:fill="D0DDEF"/><w:tblLayout w:type="fixed"/><w:tblLook w:val="04A0" w:firstRow="1" w:lastRow="0" w:firstColumn="1" w:lastColumn="0" w:noHBand="0" w:noVBand="1"/></w:tblPr><w:tblGrid><w:gridCol w:w="5910"/><w:gridCol w:w="3094"/></w:tblGrid><w:tr w14:paraId="4668C755" w14:textId="77777777" w:rsidTr="00085F35"><w:trPr><w:cantSplit/><w:trHeight w:val="2941"/></w:trPr><w:tc><w:tcPr><w:tcW w:w="5910" w:type="dxa"/><w:tcBorders><w:top w:val="nil"/><w:left w:val="nil"/><w:right w:val="nil"/></w:tcBorders><w:shd w:val="clear" w:color="auto" w:fill="auto"/><w:tcMar><w:top w:w="80" w:type="dxa"/><w:left w:w="80" w:type="dxa"/><w:bottom w:w="80" w:type="dxa"/><w:right w:w="80" w:type="dxa"/></w:tcMar></w:tcPr><w:tbl><w:tblPr><w:tblStyle w:val="Tabelraster"/><w:tblW w:w="0" w:type="auto"/><w:tblBorders><w:top w:val="none" w:sz="0" w:space="0" w:color="auto"/><w:left w:val="none" w:sz="0" w:space="0" w:color="auto"/><w:bottom w:val="none" w:sz="0" w:space="0" w:color="auto"/><w:right w:val="none" w:sz="0" w:space="0" w:color="auto"/><w:insideH w:val="none" w:sz="0" w:space="0" w:color="auto"/><w:insideV w:val="none" w:sz="0" w:space="0" w:color="auto"/></w:tblBorders><w:tblLayout w:type="fixed"/><w:tblLook w:val="04A0" w:firstRow="1" w:lastRow="0" w:firstColumn="1" w:lastColumn="0" w:noHBand="0" w:noVBand="1"/></w:tblPr><w:tblGrid><w:gridCol w:w="1790"/><w:gridCol w:w="3950"/></w:tblGrid><w:tr w14:paraId="60F0F608" w14:textId="77777777" w:rsidTr="00085F35"><w:tc><w:tcPr><w:tcW w:w="1790" w:type="dxa"/></w:tcPr><w:p w14:paraId="44A05D2A" w14:textId="77777777" w:rsidRDefault="00226DE9" w:rsidP="00085F35"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:keepLines/><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr></w:pPr></w:p><w:p w14:paraId="10CC7F18" w14:textId="6964370A" w:rsidRDefault="00825312" w:rsidP="00085F35"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:keepLines/><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr></w:pPr></w:p></w:tc><w:tc><w:tcPr><w:tcW w:w="3950" w:type="dxa"/></w:tcPr><w:p w14:paraId="1034665B" w14:textId="77777777" w:rsidRDefault="00226DE9" w:rsidP="00085F35"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:keepLines/><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr></w:pPr></w:p><w:p w14:paraId="35C1D00A" w14:textId="0FC1D1D7" w:rsidRDefault="00825312" w:rsidP="00825312"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:keepLines/><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr></w:pPr><w:r><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr><w:t>[opportunities.opportunity_id]</w:t></w:r><w:r><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr><w:br/></w:r><w:r><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr><w:t>[opportunities.testje_field_sugar_image]</w:t></w:r></w:p><w:p w14:paraId="6F52B133" w14:textId="77777777" w:rsidRDefault="00825312" w:rsidP="00825312"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:keepLines/><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr></w:pPr><w:r><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr><w:t>[opportunities.testje_id_sugar_image]</w:t></w:r></w:p><w:p w14:paraId="45477219" w14:textId="410205AB" w:rsidRDefault="00825312" w:rsidP="00825312"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:keepLines/><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr></w:pPr><w:r><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr><w:t>[opportunities.testje_module_sugar_image]</w:t></w:r></w:p></w:tc></w:tr></w:tbl><w:p w14:paraId="171B7FA7" w14:textId="77777777" w:rsidRDefault="00226DE9" w:rsidP="00085F35"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:keepLines/><w:spacing w:after="0" w:line="240" w:lineRule="auto"/></w:pPr></w:p></w:tc><w:tc><w:tcPr><w:tcW w:w="3094" w:type="dxa"/><w:tcBorders><w:top w:val="nil"/><w:left w:val="nil"/><w:right w:val="nil"/></w:tcBorders><w:shd w:val="clear" w:color="auto" w:fill="auto"/><w:tcMar><w:top w:w="80" w:type="dxa"/><w:left w:w="80" w:type="dxa"/><w:bottom w:w="80" w:type="dxa"/><w:right w:w="80" w:type="dxa"/></w:tcMar></w:tcPr><w:p w14:paraId="14FCEEFB" w14:textId="77777777" w:rsidRDefault="00226DE9" w:rsidP="00085F35"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:keepLines/><w:spacing w:after="0"/></w:pPr><w:r><w:drawing><wp:anchor distT="0" distB="0" distL="114300" distR="114300" simplePos="0" relativeHeight="251659264" behindDoc="0" locked="0" layoutInCell="1" allowOverlap="1" wp14:anchorId="5866F6BA" wp14:editId="0F704819"><wp:simplePos x="4770408" y="1199072"/><wp:positionH relativeFrom="margin"><wp:align>right</wp:align></wp:positionH><wp:positionV relativeFrom="margin"><wp:align>top</wp:align></wp:positionV><wp:extent cx="1863090" cy="1863090"/><wp:effectExtent l="0" t="0" r="3810" b="3810"/><wp:wrapSquare wrapText="bothSides"/><wp:docPr id="7" name="Picture 7" descr=""/><wp:cNvGraphicFramePr><a:graphicFrameLocks xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main" noChangeAspect="1"/></wp:cNvGraphicFramePr><a:graphic xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main"><a:graphicData uri="http://schemas.openxmlformats.org/drawingml/2006/picture"><pic:pic xmlns:pic="http://schemas.openxmlformats.org/drawingml/2006/picture"><pic:nvPicPr><pic:cNvPr id="7" name="Picture 7" descr=""/><pic:cNvPicPr/></pic:nvPicPr><pic:blipFill><a:blip r:embed="rId4" cstate="print"><a:extLst><a:ext uri="{28A0092B-C50C-407E-A947-70E740481C1C}"><a14:useLocalDpi xmlns:a14="http://schemas.microsoft.com/office/drawing/2010/main" val="0"/></a:ext></a:extLst></a:blip><a:stretch><a:fillRect/></a:stretch></pic:blipFill><pic:spPr><a:xfrm><a:off x="0" y="0"/><a:ext cx="1863090" cy="1863090"/></a:xfrm><a:prstGeom prst="rect"><a:avLst/></a:prstGeom></pic:spPr></pic:pic></a:graphicData></a:graphic></wp:anchor></w:drawing></w:r></w:p></w:tc></w:tr></w:tbl><w:p w14:paraId="3CB3F1E7" w14:textId="77777777" w:rsidRDefault="004C44CD" w:rsidP="00226DE9"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr></w:pPr><w:r><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/><w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto"/></w:rPr><w:pict w14:anchorId="79B6559E"><v:rect id="_x0000_i1025" alt="" style="width:453.6pt;height:1.5pt;mso-width-percent:0;mso-height-percent:0;mso-position-vertical:absolute;mso-width-percent:0;mso-height-percent:0" o:hralign="center" o:hrstd="t" o:hr="t" fillcolor="#a0a0a0" stroked="f"/></w:pict></w:r></w:p><w:p w14:paraId="5E540594" w14:textId="687D0A72" w:rsidRDefault="00226DE9" w:rsidP="00226DE9"><w:pPr><w:pStyle w:val="Hoofdtekst"/><w:rPr><w:rFonts w:ascii="Verdana" w:eastAsia="Verdana" w:hAnsi="Verdana" w:cs="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr></w:pPr><w:r><w:rPr><w:rFonts w:ascii="Verdana" w:hAnsi="Verdana"/><w:sz w:val="18"/><w:szCs w:val="18"/></w:rPr><w:t>';
$new = 950271;
$offset = 6675 - strlen("[opportunities.testje_field_sugar_image;ope=changepic;tagpos=inside;adjust]"); #;
$len = 7;

echo substr_replace($text, $new , $offset, $len);
`

But no clue yet how to fix it correctly.

**Comment by drentjess** [comment]
Exactly, very glad to have a proper workaround for now. However it would be huge to see this fixed in a future release.

**Comment by Skrol29** [comment]
This bug should be fixed with OpenTBS version 1.11.2.
